internal/schema/schema.go

```diff
@@ -74,21 +77,49 @@ func NewEphemeral(resolver Resolver) Manager {
 	return mgr
 }
 
-func (m *manager) LoadSchema(ctx context.Context, url string) (*jsonschema.Schema, error) {
-	entry, ok := m.cache.Get(url)
+func (m *manager) LoadSchema(ctx context.Context, _url string) (*jsonschema.Schema, error) {
+	entry, ok := m.cache.Get(_url)
 	if ok {
 		return entry.schema, entry.err
 	}
 
 	e := &cacheEntry{}
-	e.schema, e.err = m.loadSchemaFromStore(ctx, url)
+	e.schema, e.err = m.loadSchemaFromStore(ctx, _url)
+	if e.err == nil || !errors.Is(e.err, fs.ErrNotExist) {
+		m.cache.Set(_url, e)
+		return e.schema, e.err
+	}
 
-	if e.err != nil && errors.Is(e.err, fs.ErrNotExist) {
-		e.err = fmt.Errorf("schema %q does not exist in the store", url)
+	var jsonSchemaErr *jsonschema.SchemaError
+	var pathErr *fs.PathError
+	if !errors.As(e.err, &jsonSchemaErr) || !errors.As(e.err, &pathErr) {
+		m.cache.Set(_url, e)
+		return e.schema, e.err
 	}
 
-	m.cache.Set(url, e)
-	return e.schema, e.err
+	parsedURL, err := url.Parse(_url)
+	if err != nil {
+		e.err = fmt.Errorf("failed to parse URL: %w", err)
+		m.cache.Set(_url, e)
+		return e.schema, e.err
+	}
+	if parsedURL.Scheme != URLScheme {
+		e.err = fmt.Errorf("invalid URL scheme %q", parsedURL.Scheme)
+		m.cache.Set(_url, e)
+		return e.schema, e.err
+	}
+	parsedURLPath := strings.TrimPrefix(parsedURL.Path, "/")
+
+	pathErrPath := strings.TrimPrefix(pathErr.Path, util.SchemasDirectory+string(os.PathSeparator))
```

Comment:
You can only do this if the scheme is `cerbos://`. 
What is the actual error produced by `jsonschema` when a child schema is missing? Hopefully there's a way to extract the missing file information from the error itself.

Reply:
The `err` when the file itself is missing:
<img width="360" height="153" alt="image" src="https://github.com/user-attachments/assets/7c45e8a0-7035-458e-8ad1-3f5e5de65776" />

The `err` when the file itself is present but the file referenced is missing:
<img width="360" height="157" alt="image" src="https://github.com/user-attachments/assets/6a6df41e-37dc-4054-9283-b3baa1fe89c5" />

Reply:
Just saw @haines review above. I think he's referring to a similar idea.